schema/object.go

```diff
@@ -366,13 +367,6 @@ func (o *ObjectSchema) convertToObjectSchema(typeOrData any) (Object, bool) {
 
 func (o *ObjectSchema) validateSchemaCompatibility(schemaType Object) error {
 	fieldData := map[string]any{}
-	// Validate IDs. This is important because the IDs should match.
```

Comment:
So ... now object IDs needn't match. But what do we lose here? What was the actual point to this check in the first place? Is there some compatibility check we need to build another way?

Reply:
As we discussed in SUM, the effect of this is to replace strong typing with "duck typing".  This could be problematic if, for example the object in question contained five properties which were all strings, there would be no way to detect type incompatibility if the upstream object were changed to present the strings in a different order; if we were to check the ID, then the upstream definition could provide a modified ID in the new definition, and then the incompatibility would be evident.

However, retaining the ID check means that there has to be a way for the ID defined in the main workflow to match the ID (for the same object schema) defined in the subworkflow, and I don't think that that is possible as the code stands today.  So, addressing this problem would mean modifying the ID generation instead of removing the check.